src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1644,9 +1643,7 @@ private void ClearCachedAnalysisDataIfAnalyzed(SyntaxReference declaration, ISym
             ComputeDeclarationsInNode(semanticModel, symbol, declaringReferenceSyntax, topmostNodeForAnalysis, builder, cancellationToken);
 
             var isPartialDeclAnalysis = analysisScope.FilterSpanOpt.HasValue && !analysisScope.ContainsSpan(topmostNodeForAnalysis.FullSpan);
-            var nodesToAnalyze = shouldExecuteSyntaxNodeActions ?
```

Comment:
We can actually remove this cache in future- this was added due to RPS issues from v1 engine running separately for each closed document.